cmake/multi_image.cmake

```diff
@@ -215,6 +215,9 @@ function(add_child_image_from_source)
     EXTRA_KCONFIG_TARGETS
     NCS_TOOLCHAIN_VERSION
     PM_DOMAINS
+    CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC
+    CONFIG_CLOCK_CONTROL_NRF_K32SRC_XTAL
+    CONFIG_CLOCK_CONTROL_NRF_K32SRC_SYNTH
```

Comment:
A couple of issues with this approach.

1) The commit message mention nRF5340 specifically, but those settings exists for all SoCs.
    If it is only nRF5340 app core --> netcore that needs this settings in sync, then this seems wrong as it will impact all SoCs.
2) This will impact all child images, spm, rpmsg, rpc, mcuboot, and not only the netcore child image.
    Is that really correct ?
3) The passing of those settings are intended for CMake build variables.
    Kconfig settings should not go through this, as using Kconfig settings here will not work as expected.

    a) If user changes clock control in parent image, for example using `menuconfig` then the updated setting will *not* be propagated to the child image(s)
    b) User can still change the child image independently, for example calling `ninja hci_rpmsg_menuconfig` and then reconfigure the netcore image without the parent image being aware.

Sounds like we need to come up with a proper solution where a parent image can enforce a Kconfig child image setting and disable this setting from re-configuration in the child, while at the same time ensuring that parent re-conf will also be proagated.

Reply:
Please open an improvement task for followup with proper description on what we precisely need.

Reply:
Yes I'll do that, because we also plan to move clock config to devicetree in the future if I'm not mistaken. It would be good to consider to sync/lock the two domains' clock config together when we do this refactoring.